tuxemon/states/combat/combat_classes.py

```diff
@@ -0,0 +1,164 @@
+            return self._sprites[method]
+        except KeyError:
+            sprite = self.load_method_animation(method, is_flipped)
+            self._sprites[method] = sprite
+            return sprite
+
+    @staticmethod
+    def load_method_animation(
+        method: Union[Technique, Condition, Item], is_flipped: bool
+    ) -> Optional[Sprite]:
+        """
+        Return animated sprite from a technique, condition or item.
+
+        Parameters:
+            method: Whose sprite is requested.
+            is_flipped: Flag to determine whether animation frames should be flipped.
+
+        Returns:
+            Sprite associated with the animation.
+
+        """
+        if not method.animation:
+            return None
+
+        ani = AnimationEntity(method.animation)
+        if is_flipped:
+            ani.play.flip(method.flip_axes)
+        return Sprite(animation=ani.play)
+
+
+class EnqueuedAction(NamedTuple):
+    user: Union[Monster, NPC, None]
+    method: Union[Technique, Item, Condition, None]
+    target: Monster
+
+
+class ActionQueue:
+
+    def __init__(self) -> None:
+        self._action_queue: list[EnqueuedAction] = []
+
+    @property
+    def queue(self) -> list[EnqueuedAction]:
+        """Returns the current action queue."""
+        return self._action_queue
+
+    def enqueue(self, action: EnqueuedAction) -> None:
+        """Adds an action to the end of the queue."""
+        self._action_queue.append(action)
+
+    def dequeue(self, action: EnqueuedAction) -> None:
+        """Removes an action from the queue if it exists."""
+        try:
+            self._action_queue.remove(action)
+        except ValueError:
+            raise ValueError(f"Action {action} not found in queue")
+
+    def pop(self) -> EnqueuedAction:
+        """Removes and returns the last action from the queue."""
+        return self._action_queue.pop()
+
+    def is_empty(self) -> bool:
+        """Returns True if the queue is empty, False otherwise."""
+        return len(self._action_queue) == 0
+
+    def clear_queue(self) -> None:
+        """Clears the entire queue."""
+        self._action_queue.clear()
+
+    def sort(self) -> None:
+        """Sorts the queue based on the action's sort key (game rules).
+        * Techniques that damage are sorted by monster speed
+        * Items are sorted by trainer speed
+        """
+
+        def get_action_sort_key(action: EnqueuedAction) -> tuple[int, int]:
+            if action.method is None:
+                return 0, 0
+            else:
+                action_sort_type = action.method.sort
+                primary_order = prepare.SORT_ORDER.index(action_sort_type)
+
+                if action_sort_type in ["meta", "potion"]:
+                    return primary_order, 0
+                else:
+                    if action.user is None:
+                        return 0, 0
+                    else:
+                        return primary_order, action.user.speed_test(action)
```

Comment:
`                        return primary_order, -action.user.speed_test(action)`

I think if you do this you don't need the next section

Reply:
I think it would be clearer if you applied the negative to the primary_order, and left the reverse=True out.

Reply:
I'm explaining it, even if it's worthless and a waste of time, but you're right, because by negating the result (-), we are sorting the action with the same primary order in descending order based on the speed test result, so we can remove it